src/js/services/ng-rankings.js

```diff
@@ -0,0 +1,96 @@
+/**
+ * This is a service that can calculate the rankings.
+ */
+define('services/ng-rankings',[
+    'services/ng-services',
+    'services/ng-stages',
+    'services/ng-teams',
+    'services/ng-score'
+],function(module) {
+    "use strict";
+
+    return module.service('$rankings',
+        ['$stages','$teams', '$score',
+        function($stages, $teams, $score) {
+
+            const EMPTY = '---';
+
+            function group(arr, func) {
+                return arr.reduce((groups, item) => {
+                    let key = func(item);
+                    if(!groups.hasOwnProperty(key)) {
+                        groups[key] = [];
+                    }
+                    groups[key].push(item);
+                    return groups;
+                }, {});
+            }
+
+            function multigroup(arr, funcs) {
+                let currFunc = funcs[0];
+                let result = group(arr, currFunc);
+                if(funcs.length > 1) {
+                    let slicedFuncs = funcs.slice(1);
+                    for(let key in result) {
+                        result[key] = multigroup(result[key], slicedFuncs);
+                    }
+                }
+                return result;
+            }
+
+            function compareRanks(rank1, rank2) {
+                let sortedRank1Scores = rank1
+            }
+
+            function Rank(rank, team, stage) {
+                this.team = team;
+                this.stage = stage;
+                this.empty = EMPTY;
+
+                this.scores = new Array(stage.rounds).fill('score').map((u,i) => {
```

Comment:
Because if I don't fill it, the map function will ignore it (it ignores undefined values)